client/src/main/java/sjt/http/client/module/request/HttpRequest.java

```diff
@@ -0,0 +1,134 @@
+package sjt.http.client.module.request;
+
+import sjt.http.client.module.HttpMethod;
+
+import java.io.BufferedOutputStream;
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.OutputStream;
+import java.util.Map;
+
+public class HttpRequest {
+
+    private static final String HEADER_DELIMITER = ": ";
+
+    private RequestLine requestLine;
+    private Map<String, String> requestHeader;
+    private String requestBody;
+
+    public HttpRequest (HttpMethod httpMethod, String uri, String version) {
+        requestLine = new RequestLine(httpMethod, uri, version);
+    }
+
+    /**
+     * BufferedReader로 부터 request 읽어오기
+     *
+     * @param reader
+     * @throws IOException
+     */
+    public void readHttpRequest(BufferedReader reader) throws IOException {
+        requestLine = new RequestLine(reader.readLine());
```

Comment:
생성자를 활용하는 것도 좋지만, 좀 더 의도를 파악하기 쉽도록 static factory method 를 구현해 활용해 보는 것은 어떨까요?
```suggestion
        requestLine = RequestLine.parse(reader.readLine());
```
참고: https://docs.oracle.com/javase/tutorial/datetime/overview/naming.html

Reply:
넵  tatic factory method을 활용한게 더 명확한 것 같아요! 감사합니다 :)